packages/dd-trace/src/appsec/remote_config/index.js

```diff
@@ -23,9 +24,32 @@ function enable (config, appsec) {
       rc.updateCapabilities(RemoteConfigCapabilities.ASM_ACTIVATION, true)
     }
 
-    rc.setProductHandler('ASM_FEATURES', (action, rcConfig) => {
+    rc.updateCapabilities(RemoteConfigCapabilities.ASM_AUTO_USER_INSTRUM_MODE, true)
+
+    let autoUserInstrumModeId
+
+    rc.setProductHandler('ASM_FEATURES', (action, rcConfig, configId) => {
       if (!rcConfig) return
 
+      // this is put before other handlers because it can reject the config
+      if (typeof rcConfig.auto_user_instrum?.mode === 'string') {
+        if (action === 'apply' || action === 'modify') {
+          // check if there is already a config applied with this field
+          if (autoUserInstrumModeId && configId !== autoUserInstrumModeId) {
```

Comment:
this means that once a instrum mode is received, it cannot be changed, no?

Reply:
are you sure ? doesn't it mean that if a instrum mode is received, it needs to be unnaplied before a new one can be applied. But in the case of a `action === 'modify`, the configId is the say, so it should pass the check

Reply:
so, when a value changes in the UI, there is first an unapply action and then after that another one to apply the new value?
sorry don't know the details of how RC works.

Reply:
idk but it doesn't have to be, are you talking about the action === 'modify' case ?